OXRS-BMD-SmokeDetector-ESP32-FW.ino

```diff
@@ -712,7 +712,7 @@ void initialiseMcp(int mcp, int address)
     mcp23017[mcp].begin_I2C(address);
     for (uint8_t pin = 0; pin < MCP_PIN_COUNT; pin++)
     {
-      mcp23017[mcp].pinMode(pin, address == MCP_INPUT_ADDR ? INPUT : OUTPUT);
+      mcp23017[mcp].pinMode(pin, address == MCP_INPUT_ADDR ? INPUT_PULLUP : OUTPUT);

```

Comment:
This needs to be reverted - the smoke detector hardware has h/w pullups built in so this shouldn't be in the firmware.